### ISSUE TITLE
python client read_with_poll not working as expected

### ISSUE THREAD

**Issue by elvizlai** [tembo-io/pgmq#357]
```
read_messages = queue.read_with_poll(
    "my_queue", vt=30, qty=500, max_poll_seconds=1000, poll_interval_ms=100
)
for message in read_messages:
    print(message)
```
For this scenario what we want should read_messages should be length=500 or the `read_with_poll` should be blocked util 1000 seconds.

But what I got was, it immediately print all 20 messages(all message to my_queue) and then exit.

**Comment by brianpursley** [comment]
I think this is the intended behavior.

From the [docs](https://tembo.io/pgmq/api/sql/functions/#read_with_poll):

> ...When there are no messages in the queue, the function call will wait for max_poll_seconds in duration before returning. If messages reach the queue during that duration, they will be read and returned immediately.

So `qty` is the maximum number of messages to read at once.  In your case, it will read *up to* 500 messages, and if no message arrives in 1000 seconds, it will timeout.

If I understand the code correctly, it uses `qty` as a limit (line 131) and if there are any results at all, it returns out of the loop (line 151):
https://github.com/tembo-io/pgmq/blob/793ecd625b49c02df6a449bcb0301f085daf8ea0/pgmq-extension/sql/pgmq.sql#L115-L155

The docs for `qty` could probably use some clarification about this.  Currently it says:
> The number of messages to read from the queue. Defaults to 1.

but should probably be more like:
> The **maximum** number of messages to read from the queue. Defaults to 1.

**Comment by elvizlai** [comment]
Yes, maybe the repo markdown doc misleading

https://github.com/tembo-io/pgmq/blob/793ecd625b49c02df6a449bcb0301f085daf8ea0/tembo-pgmq-python/README.md?plain=1#L138

until it either finds the specified number of messages (`qty`) or the `max_poll_seconds` duration is reached

**Comment by v0idpwn** [comment]
Yes, it's expected behaviour, and the doc there is wrong. PRs welcome :)
